File: BinaryConverterJavaWebServer/src/java/ist/enesuysal/thesis/Message.java
package ist.enesuysal.thesis;

import java.io.IOException;
import java.util.Arrays;
import java.util.logging.Level;
import java.util.logging.Logger;
import sun.misc.BASE64Decoder;
import sun.misc.BASE64Encoder;

public class Message {

    int messageInt;
    String messageString;
    Boolean messageBool;
    Object messageObject;
    byte[] messageSerilized= new byte[0];

    public Message(byte message) {
        messageSerilized = new byte[message];
    }
    public Message(int message) {
        messageSerilized = CentralSerializer.serializePrimitive(int.class, message, messageSerilized);//convertToByteArray(message,messageSerilized);
       
    }

    public Message(String message) {
        messageSerilized = CentralSerializer.convertToByteArray(message, messageSerilized);
    }

    public Message(boolean message) {
        messageSerilized = CentralSerializer.convertToByteArray(message, messageSerilized);
    }
    
    public Message(char message) {
        messageSerilized = CentralSerializer.convertToByteArray(message,messageSerilized);
    }
     public Message(long message) {
        messageSerilized = CentralSerializer.convertToByteArray(message,messageSerilized);
    }
      public Message(short message) {
        messageSerilized = CentralSerializer.convertToByteArray(message,messageSerilized);
    }
     public Message(float message) {
        messageSerilized = CentralSerializer.convertToByteArray(message,messageSerilized);
    }
     public Message(double message) {
        messageSerilized = CentralSerializer.convertToByteArray(message,messageSerilized);
    }
    public Message(Object message) throws IllegalArgumentException, IllegalAccessException, Exception {
        messageSerilized = CentralSerializer.objectToByteArray(message, messageSerilized);
    }

    public String Seriliaze() {
        byte[] arrayByte = null;
        BASE64Encoder encoder = new BASE64Encoder();
        String temp_inBase64 = encoder.encode(messageSerilized);
        return temp_inBase64;
    }

    public byte[] DeSeriliaze(String temp_inBase64) {
        byte[] arrayByte = null;
        BASE64Decoder decoder = new BASE64Decoder();
        try {
            arrayByte = decoder.decodeBuffer(temp_inBase64);
        } catch (IOException ex) {
            Logger.getLogger(Message.class.getName()).log(Level.SEVERE, null, ex);
        }
        return arrayByte;
    }
}

File: CSharpWebServer/CSharpWebServer/ist.enesuysal.thesis/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSharpWebServer.ist.enesuysal.thesis
{
    public class Message
    {
        int messageInt;
        String messageString;
        Boolean messageBool;
        Object messageObject;
        byte[] messageSerilized = new byte[0];
        public Message(byte message)
        {
            messageSerilized = CentralSerializer.serializePrimitive(typeof(byte), message, messageSerilized);
        }
        public Message(string message)
        {
            messageSerilized = CentralSerializer.serializePrimitive(typeof(string), message, messageSerilized);
        }
      
        public Message(int message)
        {
             messageSerilized = CentralSerializer.serializePrimitive(typeof(int), message, messageSerilized);//convertToByteArray(message,messageSerilized);

        }
        public Message(bool message)
        {
            messageSerilized = CentralSerializer.serializePrimitive(typeof(bool), message, messageSerilized);
        }

        public Message(char message)
        {
            messageSerilized = CentralSerializer.serializePrimitive(typeof(char), message, messageSerilized);
        }
        public Message(long message)
        {
            messageSerilized = CentralSerializer.serializePrimitive(typeof(long), message, messageSerilized);
        }
        public Message(short message)
        {
            messageSerilized = CentralSerializer.serializePrimitive(typeof(short), message, messageSerilized);
        }
        public Message(float message)
        {
            messageSerilized = CentralSerializer.serializePrimitive(typeof(float), message, messageSerilized);
        }
        public Message(double message)
        {
            messageSerilized = CentralSerializer.serializePrimitive(typeof(double), message, messageSerilized);
        }

        public string Seriliaze()
        {
            string temp_inBase64 = Convert.ToBase64String(messageSerilized);
            return temp_inBase64;
        } 
    }
}
